### PR TITLE
Switch error messages to new translation system

### PR DESCRIPTION
**Pull request by AlexEven**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | The former Tools::displayError('Blabla') syntax prevented such strings to be parsed by the new translation system. So we changed it to have the new trans method. And of course added domains and updated some variables at the same time. It's only for admin controllers so far. |
| Type? | improvement |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | n/a |
| How to test? | See if these errors are still displayed + should be in Crowdin once it's all updated. |

And.... good luck for the review!! :grimacing:

### CODE REVIEW

**Comment by maximebiloe** [comment]
infinite

File: controllers/admin/AdminCategoriesController.php [hunk 1]
```diff
@@ -871,13 +871,13 @@ public function processFatherlessProducts($id_parent)
     public function processPosition()
     {
         if ($this->access('edit') !== '1') {
-            $this->errors[] = Tools::displayError('You do not have permission to edit this.');
+            $this->errors[] = $this->trans('You do not have permission to edit this.', array(), 'Admin.Notifications.Error');
         } elseif (!Validate::isLoadedObject($object = new Category((int)Tools::getValue($this->identifier, Tools::getValue('id_category_to_move', 1))))) {
-            $this->errors[] = Tools::displayError('An error occurred while updating the status for an object.').' <b>'.
-                $this->table.'</b> '.Tools::displayError('(cannot load object)');
+            $this->errors[] = $this->trans('An error occurred while updating the status for an object.', array(), 'Admin.Notifications.Error').' <b>'.
```
Comment (maximebiloe) [inline controllers/admin/AdminCategoriesController.php]: This string can be optimized for RTL with variables.

File: controllers/admin/AdminCmsCategoriesController.php [hunk 2]
```diff
@@ -121,14 +121,14 @@ public function postProcess()
                         $identifier = ((int)$object->id_parent ? '&id_cms_category='.(int)$object->id_parent : '');
                         Tools::redirectAdmin(self::$currentIndex.'&conf=5'.$identifier.'&token='.Tools::getValue('token'));
                     } else {
-                        $this->errors[] = Tools::displayError('An error occurred while updating the status.');
+                        $this->errors[] = $this->trans('An error occurred while updating the status.', array(), 'Admin.Notifications.Error');
                     }
                 } else {
-                    $this->errors[] = Tools::displayError('An error occurred while updating the status for an object.')
-                        .' <b>'.$this->table.'</b> '.Tools::displayError('(cannot load object)');
+                    $this->errors[] = $this->trans('An error occurred while updating the status for an object.', array(), 'Admin.Notifications.Error')
```
Comment (maximebiloe) [inline controllers/admin/AdminCmsCategoriesController.php]: It can be optimized too

File: controllers/admin/AdminCmsCategoriesController.php [hunk 3]
```diff
@@ -137,8 +137,8 @@ public function postProcess()
                 if (Validate::isLoadedObject($object = $this->loadObject()) && isset($this->fieldImageSettings)) {
                     // check if request at least one object with noZeroObject
                     if (isset($object->noZeroObject) && count($taxes = call_user_func(array($this->className, $object->noZeroObject))) <= 1) {
-                        $this->errors[] = Tools::displayError('You need at least one object.')
-                            .' <b>'.$this->table.'</b><br />'.Tools::displayError('You cannot delete all of the items.');
+                        $this->errors[] = $this->trans('You need at least one object.', array(), 'Admin.Notifications.Error')
```
Comment (maximebiloe) [inline controllers/admin/AdminCmsCategoriesController.php]: Same thing here

File: controllers/admin/AdminRequestSqlController.php [hunk 4]
```diff
@@ -382,42 +382,69 @@ public function displayError($e)
...
+                                '%attribute%' => $e[$key]['attribut'][0],
+                                '%table%' => $e[$key]['attribut'][1],
+                            ),
+                            'Admin.Parameters.Notification'
                         );
                     } elseif (isset($e[$key]['*'])) {
-                        $this->errors[] = Tools::displayError('The "*" operator cannot be used in a nested query.');
+                        $this->errors[] = $this->trans('The "*" operator cannot be used in a nested query.', array(), 'Admin.Parameters.Notification');
                     } else {
                         $this->errors[] = Tools::displayError('Undefined "checkedSelect" error');
```
Comment (maximebiloe) [inline controllers/admin/AdminRequestSqlController.php]: Don't you forget this one?

**Comment by maximebiloe** [comment]
I think your forgot some in this file `AdminRequestSqlController `

**Comment by AlexEven** [comment]
@maximebiloe : yes, I ignored them on purpose. I'm not sure it's valuable to the merchant and really need to be translated :)
Thank you!

**Comment by mickaelandrieu** [comment]
Go for it, it will take less time to fix errors than do all the tests to check if everything is fine.

Thank you @AlexEven

**Pull request merged**